lib/datagrid/translation.rb

```diff
@@ -0,0 +1,25 @@
+require "action_view/helpers"
+
+module Datagrid
+  module Translation
+    include ActionView::Helpers::TranslationHelper
+
+    def translate_from_namespace(namespace, grid_class, key)
+      deprecated_translation_key = :"#{grid_class.param_name}.#{namespace}.#{key}"
+
+      if grid_class.name.include?("::") && I18n.exists?(deprecated_translation_key)
```

Comment:
Some people overwrite `param_name` at the instead level making it prettier or shorter like `def self.param_name; 'grid';end`. So the deprecated usage not necessarily mean there is a name space. We should check if `i18n_key != param_name` instead.